velox/connectors/hive/tests/HivePartitionUtilTest.cpp

```diff
@@ -117,3 +117,47 @@ TEST_F(HivePartitionUtilTest, partitionName) {
         "Unsupported partition type: MAP");
   }
 }
+
+TEST_F(HivePartitionUtilTest, partitionNameForNull) {
+  std::vector<std::string> partitionColumnNames{
+      "flat_bool_col",
+      "flat_tinyint_col",
+      "flat_smallint_col",
+      "flat_int_col",
+      "flat_bigint_col",
+      "dict_string_col",
+      "const_date_col"};
+
+  RowVectorPtr input = makeRowVector(
+      partitionColumnNames,
+      {makeNullableFlatVector<bool>(
+           std::vector<std::optional<bool>>{std::nullopt}),
+       makeNullableFlatVector<int8_t>(
+           std::vector<std::optional<int8_t>>{std::nullopt}),
+       makeNullableFlatVector<int16_t>(
+           std::vector<std::optional<int16_t>>{std::nullopt}),
+       makeNullableFlatVector<int32_t>(
+           std::vector<std::optional<int32_t>>{std::nullopt}),
+       makeNullableFlatVector<int64_t>(
+           std::vector<std::optional<int64_t>>{std::nullopt}),
+       makeNullableFlatVector<StringView>(
+           std::vector<std::optional<StringView>>{std::nullopt}),
+       makeConstant<int32_t>(std::nullopt, 1, DATE())});
+
+  for (auto i = 1; i <= partitionColumnNames.size(); i++) {
```

Comment:
What's the point of testing 1, 2, 3, ...N partition keys? Why not just test all partition key types, but use single partition key for each test?

Reply:
it follows previous test, I changed to use single partition key for each test.